Make it compatible with ActiveModel 5

ahukkanen opened the pull request:
- Bumps activemodel version to 5
- Bumps bundler version to 2
- Adds a new integration test to test the HetuValidator

ahukkanen committed:
Bump bundler and activemodel versions + add integration test

- Bumps activemodel version to 5
- Bumps bundler version to 2
- Adds a new integration test to test the HetuValidator

ahukkanen commented:
Is there something holding this back? Maybe we can find a solution in case the concerns are raised?

Or should we make it so that it would work for both `4.x` AND `5.x` at the same time? There is not any incompatibilities regarding this library between those two versions. This could, however, potentially break for future versions...

alexhanh commented:
Yes. It should work for 4.x as I bet it is still in widespread use, although I couldn’t find statistics to back this claim up.

ahukkanen committed:
Allow multiple mainline versions of dependencies

For more widespread use, the same version can now be used with
multiple mainline versions of the dependencies.

ahukkanen committed:
Test code refactoring

- Wrap the PersonModel class used in the validator test to the
  `Henkilotunnus::Test` module
- Lint the code with rubocop

ahukkanen commented:
@alexhanh I've made this change at cf699e5ab4e614f25db736f25fca1d1cea6fbfa3.

Please do note that this opens the gem for potential incompatibilities in the future in case some of the underlying APIs change. IMO it would be safer to just release a new version with the updated dependencies every time the mainline dependencies change but the current solution of course makes the gem easier to maintain.

So, either way, as you like.
